Source/SvgElementIdManager.cs

```diff
@@ -43,31 +34,59 @@ public virtual SvgElement GetElementById(string id)
 
         public virtual SvgElement GetElementById(Uri uri)
         {
-            if (uri.ToString().StartsWith("url(")) uri = new Uri(uri.ToString().Substring(4).TrimEnd(')'), UriKind.Relative);
-            if (!uri.IsAbsoluteUri && this._document.BaseUri != null && !uri.ToString().StartsWith("#"))
+            var urlString = GetUrlString(uri.ToString());
+
+            if (!urlString.StartsWith("#"))
             {
-                var fullUri = new Uri(this._document.BaseUri, uri);
-                var hash = fullUri.OriginalString.Substring(fullUri.OriginalString.LastIndexOf('#'));
-                SvgDocument doc;
-                switch (fullUri.Scheme.ToLowerInvariant())
+                var index = urlString.LastIndexOf('#');
+                var fragment = urlString.Substring(index);

```

Comment:
Is it guaranteed at this point that urlString contains "#"? Otherwise, it would throw here.

Reply:
Thank you for your review.

There are various formats of id in GetElementById methods, so same as old code.

Reply:
Ok, I see now that this has not changed - it the URL was in wrong format (e.g. missing #), it would have crashed before, too.